src/libtiled/mapwriter.cpp

```diff
@@ -443,7 +443,8 @@ void MapWriterPrivate::writeTileset(QXmlStreamWriter &w, const Tileset &tileset,
     }
 
     // Write the properties for those tiles that have them
-    for (const Tile *tile : tileset.tiles()) {
+    for (int id : tileset.sortedTileIds()) {
+        const Tile *tile = tileset.findTile(id);
         if (imageSource.isEmpty() || includeTile(tile)) {
```

Comment:
This code will only write a `tile` element if either `imageSource.isEmpty()` (when it's an image collection tileset) or when `includeTile` returns true (when the tile has been customized in some way). This will need to be adjusted such that any tile which needs its order to be saved is also included (or possibly better, when the order has been changed in any way, all tiles should be written).

It might even be alright to just remove these checks entirely since there's also a plan to store the location of each tile in the set as per #2863.

Note that the same kind of check happens in other map formats (`maptovariantconverter.cpp` and `luaplugin.cpp`).